edalize/vcs.py

```diff
@@ -0,0 +1,86 @@
+import os
+import logging
+
+from edalize.edatool import Edatool
+
+logger = logging.getLogger(__name__)
+
+class Vcs(Edatool):
+
+    tool_options = {
+        'lists' : {'vcs_options' : 'String'}
+    }
+
+    argtypes = ['plusarg', 'vlogdefine', 'vlogparam']
+
+    MAKEFILE_TEMPLATE = """
+all: $(TARGET)
+
+$(TARGET):
+	vcs -full64 -t $(TOPLEVEL) -f $(TARGET).scr -o $@ $(VCS_OPTIONS)
+
+run: $(TARGET)
+	$(TARGET) -l vcs.log $(EXTRA_OPTIONS)
+
+clean:
+	$(RM) $(TARGET)
+"""
+
+    def configure_main(self):
+        f = open(os.path.join(self.work_root, self.name+'.scr'),'w')
+
+        (src_files, incdirs) = self._get_fileset_files()
+        for key, value in self.vlogdefine.items():
+            f.write('+define+{}={}\n'.format(key, self._param_value_str(value, '')))
+
+        for key, value in self.vlogparam.items():
+            f.write('+parameter+{}.{}={}\n'.format(self.toplevel, key, self._param_value_str(value, '"')))
```

Comment:
isn't `self.toplevel` a list? At least `Edatool::__init__()` uses it that way. (Yes, the name is and was confusingly singular.) [Speaking about that, I see that I've used it as string in the spyglass backend as well... I guess we need a couple more tests here.]

Reply:
Now I look at it, I'm not even sure how you tell vcs which your top-level module is. In my other use I've just seen it tell you it has 9 (true) top-modules and then simulates all of them. I can't even find an argument in the help page to indicate and it seems to just silently accept -t (other people on the internet have mentioned -top, which also is not in the manual)